include/tvm/topi/transform.h

```diff
@@ -1608,7 +1608,11 @@ inline Tensor layout_transform(const Tensor& src, const std::string& src_layout,
       [&](const Array<Var>& dst_indices) {
         Array<PrimExpr> dst_indices_expr(dst_indices.begin(), dst_indices.end());
         Array<PrimExpr> src_indices = layout_converter.BackwardIndex(dst_indices_expr);
-        return src(src_indices);
+        PrimExpr in_range = PrimExpr(1) > PrimExpr(0);  // init with dtype=bool and value=true
+        for (size_t i = 0; i < src.ndim(); ++i) {
+          in_range = in_range && (src_indices[i] < src->shape[i]);
+        }
+        return if_then_else(in_range, src(src_indices), tvm::cast(src->dtype, PrimExpr(0)));
```

Comment:
Why do you need this change? Returning 0 for out of bounds doesn't seem right to me... Is there a "valid" out of bounds for which returning 0 is the right thing to do?

Reply:
This is similar to zero padding. For example, when a tensor with shape=[ 64, 3, 7, 7, ] is going to be transformed from "OIHW" to "OHWI16o2i", it's new shape=[ 4, 7, 7, **2**, 16, **2**,  ], which is bigger then the original tensor in total.

Reply:
ah ok, makes sense. Since enlarged region is zero, it doesn't modify the result.